lib/device_detector/version_extractor.rb

```diff
@@ -18,13 +18,17 @@ def extract_version
     end
 
     def replace_version_string_with(match_data)
-      string = version_string
+      string = version_string.to_s
 
       1.upto(9) do |index|
-        string = string.gsub(/\$#{index}/, match_data[index]) if match_data[index]
+        if match_data[index]
+          string = string.gsub(/\$#{index}/, match_data[index])
+        else
+          string = string.gsub(/\$#{index}/, '')
+        end
       end
 
-      string
+      string.strip
```

Comment:
Yes, for regexes like:
```yaml
- regex: '(?:Apple-)?iPhone ?(3GS?|4S?|5[CS]?|6(:? Plus)?)?'
  model: 'iPhone $1'
  device: 'smartphone'
```

If the first match is made but nothing for the capture group, the output will be `'iPhone '` (empty trailing space at the end).